### PR TITLE
Document new contact update behaviour

### PR DESCRIPTION
**Pull request by ryansch**

### CODE REVIEW

File: endpoints/contacts.md [hunk 1]
```diff
@@ -109,13 +109,79 @@ You can create a contact, email, phone, and address in one step.  Here's an exam
 
 * ```PUT /contacts/:id.json``` updates contact from parameters
 
+Contacts have four different related types: phones, emails, addresses, and web_addresses.  Additionally, phones, emails, and addresses can have primaries set.
+The update contact endpoint supports two modes of operation with regards to those related types: normal and replace. In normal mode, you can add and update existing related types. You can use replace mode to overwrite or delete existing data.
+
+You can invoke normal mode by sending the header `X-Outstand-Replace: true`.  Any other value (or the header's absence) will result in normal mode.
```
Comment (morganick) [inline endpoints/contacts.md]: *replace mode

File: endpoints/contacts.md [hunk 2]
```diff
@@ -109,13 +109,79 @@ You can create a contact, email, phone, and address in one step.  Here's an exam
...
+In replace mode the following would happen:
+- Phone id `1` would be retained
+- A new phone would be added with the number '555 555 1212'
+- Email id `10` would have its address updated
+- A new email would be added with the address `john@example.com`
+- Address id `20` is not included in the payload and would be deleted
+- A new address would be added with the above attributes
+- Web address id `30` would be retained
+
+If a related type that is also a primary is deleted (example: phone id `1` is also the primary_phone_id) then the primary is set to `nil` except in cases where there is only one other record which is promoted to primary (example: we have a second phone number already).
```
Comment (morganick) [inline endpoints/contacts.md]: which is *then* promoted

File: endpoints/contacts.md (hunk 1 above)
Comment (ryansch) [reply endpoints/contacts.md]: derp

**Comment by ryansch** [comment]
@morganick Fixed

**Pull request merged**